File: 프로그래머스/Level1/소수 찾기.py
# https://programmers.co.kr/learn/courses/30/lessons/12921

def solution(n):
    a = 1000000
    arr = [False, False] + [True] * (a - 1)
    m = int(a ** 0.5)
    for i in range(2, m + 1):
        if arr[i]:
            for j in range(2*i, a, i):
                arr[j] = False
    answer = 0
    for i in range(2, n + 1):
        if arr[i]:
            answer += 1
    return answer

File: 프로그래머스/Level1/소수 찾기.js
// https://programmers.co.kr/learn/courses/30/lessons/12921

const solution = (n) => {
	const isPrimes = new Array(n + 1).fill(true);
	isPrimes[0] = false;
	isPrimes[1] = false;
	isPrimes.forEach((n, idx) => {
		if (n === true) {
			for (let i = 2 * idx; i < isPrimes.length; i += idx) {
				isPrimes[i] = false;
			}
		}
	});
	return isPrimes.filter((number) => number === true).length;
};
